File: packages/runtimeuse-client-python/src/runtimeuse_client/exceptions.py
import json


class CancelledException(Exception):
    """Raised when an agent runtime invocation is cancelled."""

    pass


class AgentRuntimeError(Exception):
    """Raised when the agent runtime returns an error message."""

    def __init__(self, error: str, metadata: dict | None = None):
        super().__init__(error)
        self.error = error
        self.metadata = metadata

    def __str__(self) -> str:
        if not self.metadata:
            return self.error

        try:
            metadata_str = json.dumps(self.metadata, sort_keys=True, default=str)
        except TypeError:
            metadata_str = str(self.metadata)

        return f"{self.error}\nmetadata: {metadata_str}"

File: packages/runtimeuse-client-ts/src/exceptions.ts
export class CancelledException extends Error {
  constructor(message = "Query was cancelled") {
    super(message);
    this.name = "CancelledException";
  }
}

export class AgentRuntimeError extends Error {
  readonly error: string;
  readonly metadata?: Record<string, unknown>;

  constructor(error: string, metadata?: Record<string, unknown>) {
    let msg = error;
    if (metadata) {
      try {
        const metadataStr = JSON.stringify(metadata, Object.keys(metadata).sort());
        msg = `${error}\nmetadata: ${metadataStr}`;
      } catch {
        msg = `${error}\nmetadata: ${String(metadata)}`;
      }
    }
    super(msg);
    this.name = "AgentRuntimeError";
    this.error = error;
    this.metadata = metadata;
  }
}
